### PR TITLE
Button component

### PR DESCRIPTION
**Pull request by TanguyDucky**
Closes [DEVELOP: Button](https://app.asana.com/0/1200489836971088/1201799627233580/f) 

Figma Spec: https://www.figma.com/file/4ZukYZAMHrsCIsrZu9u2a6/Plumage-Components?node-id=92%3A38704

### CODE REVIEW

File: packages/component-library/src/components/plmg-button/plmg-button.scss [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+      @include just-text(tokens.$plmg-text-danger-strong-active);
+    }
+  }
+}
+// END Mixins
+// ------------------------------
+
+// Component style
+// ------------------------------
+button {
```
Comment (JohannesAnd) [inline packages/component-library/src/components/plmg-button/plmg-button.scss:262]: I would prefer to use a class name here, such as `.button`. This way, we only apply styles where explicitly stated. Also, if we were to extract this styling to a general-purpose library (say for Wordpress or asp.net) we could simply include the CSS and do `class="button primary medium"` and it should all work perfectly fine even outside of web components. Just a thought, as I don't know if there is a reason if this must be a HTML element for some magic to work...

**Review by JohannesAnd — COMMENTED** [review]
I am not allowed to run the test in component-library, I get the following error:
`Missing dev dependencies @types/jest@26.0.21 jest@26.6.3 jest-cli@26.6.3 puppeteer@10.0.0`

When the button is focused, it moves. I don't think this is expected behavior.

Is the button responsible for displaying the text inside it? Or should a text component be used inside? Now, it sets font and line height, but not font type. What is the plan here?

Should the button have a cursor?

**Comment by JakeH91** [comment]
:focus-visible seems to be better for the focus state:
https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible

File: packages/component-library/src/components/plmg-button/plmg-button.scss (hunk 1 above)
Comment (TanguyDucky) [reply packages/component-library/src/components/plmg-button/plmg-button.scss:262]: With Stencil and web components, each component will live in its own shadow-dom. Therefore, all styles will be isolated from each other. That's why we can be very generic and use `button`.

**Comment by TanguyDucky** [comment]
I've removed fonts from this PR in favor of https://github.com/DuckyTeam/Plumage/pull/16

**Review by JohannesAnd — APPROVED** [review]

**Pull request merged**